File: trigger.cpp
#include <iostream>
#include <fstream>
#include <signal.h>

using namespace std;

int main()
{
  int pid_var;
  char signal;

  // READ PID AND USE IT TO TRIGGER SIGNALS FOR THE PROCESS
  ifstream infile;
  
  infile.open("pid");
  infile >> pid_var;

  cout << pid_var << endl;
  cout << "Please enter l, c, or s for Lever Down, Cancel and Stop respectively" << endl;

  while(1)
  {
    cout << ">> ";
    cin >> signal;
    
    switch(signal)
    {
      case 'l': // LEVER DOWN
        kill(pid_var, SIGUSR1);
        break;
      case 'c': // CANCEL
        kill(pid_var, SIGUSR2);
        break;
      case 's': //STOP
        kill(pid_var, SIGKILL);
        break;
      default: // WRONG INPUT
        cout << "Wrong input given";
        break;
    }

  }
 
  return 0;
}

File: trigger.py
import os
import sys
import signal

# READING THE PID (GIVEN BY app_toaster.py)
f = open('pid', 'r')
pid = int(f.read())
f.close()

print pid

# READ COMMAND LINE ARGUEMENT
input = sys.argv[1]

# USE THE COOMAND LINE ARGUEMENT TO TRIGGER A SPECIFIC FUNCTION
if input == 'l':
    # LEVER DOWN / START TRIGGER
    os.kill(pid, signal.SIGUSR1)
if input == 'c':
    # CANCEL BUTTON TRIGGER
    os.kill(pid, signal.SIGUSR2)
if input == 's':
    # STOP SIGNALTRIGGER
    os.kill(pid, signal.SIGKILL)
